helix-view/src/document.rs

```diff
@@ -21,11 +21,18 @@ pub enum Mode {
     Insert,
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub enum IndentStyle {
+    Tabs,
+    Spaces(u8),
```

Comment:
Nitpick, but this could be NonZeroU8 as indent with 0 spaces does not make sense

Reply:
Yeah, the intended valid range here is [1, 8] inclusive.  Since I'm considering more than 8 also invalid, I wasn't sure if it made sense to only encode half of the validity check into the type system.  I guess it depends mainly on if NonZeroU8 is easy to work with.  I'll look into it.

Thanks!

Reply:
Oh, reading the docs for NonZeroU8, I think I see what you're getting at.  The whole enum could be a single byte, then.  I don't think this is a place where we need to worry about space, though.  Honestly, I was half-considering just going with usize here, since it's only stored once per document.